arangod/Aql/IResearchViewNode.cpp

```diff
@@ -1083,24 +1082,23 @@ IResearchViewNode::IResearchViewNode(aql::ExecutionPlan& plan,
                                                NODE_OUT_VARIABLE_PARAM)},
       _outNonMaterializedDocId{aql::Variable::varFromVPack(
           plan.getAst(), base, NODE_OUT_NM_DOC_PARAM, true)},
-      _outNonMaterializedColPtr{aql::Variable::varFromVPack(
-          plan.getAst(), base, NODE_OUT_NM_COL_PARAM, true)},
       // in case if filter is not specified
       // set it to surrogate 'RETURN ALL' node
       _filterCondition{&kAll},
       _scorers{fromVelocyPack(plan, base.get(NODE_SCORERS_PARAM))} {
-  if ((_outNonMaterializedColPtr != nullptr) !=
-      (_outNonMaterializedDocId != nullptr)) {
-    THROW_ARANGO_EXCEPTION_MESSAGE(
-        TRI_ERROR_BAD_PARAMETER,
-        absl::StrCat("invalid node config, '", NODE_OUT_NM_DOC_PARAM,
-                     "' attribute should be consistent with '",
-                     NODE_OUT_NM_COL_PARAM, "' attribute"));
-  }
   auto const viewNameSlice = base.get(NODE_VIEW_NAME_PARAM);
   auto const viewName =
       viewNameSlice.isNone() ? "" : viewNameSlice.stringView();
   auto const viewIdSlice = base.get(NODE_VIEW_ID_PARAM);
+
+  if (base.hasKey("outNmColPtr")) {
+    // Old coordinator tries to run query on
+    // new dbserver. Registers are not compatible.
+    // We must abort.
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER,
```

Comment:
Please add TODO about this condition can be removed after support last version with it (probably any 3.9 or <= 3.10.2) will be dropped

Reply:
I guess we do not prevent upgrading from EOLed versions. So this code should stay.